### PR TITLE
优化 Dockerfile

### PR DESCRIPTION
**Pull request by Coderhypo**
1. 镜像瘦身
2. 添加 pip mirror
3. manager 的 init db 和 create admin 分离

### CODE REVIEW

File: deploy/hackathon-docker/Dockerfile-server [hunk 1]
```diff
@@ -10,24 +10,24 @@ FROM python:3.6-alpine
...
-    dos2unix /opt/open-hackathon/docker-entrypoint-server.sh
+COPY ./open-hackathon-server/requirements.txt /opt/open-hackathon/open-hackathon-server
 
-RUN pip install -r /opt/open-hackathon/open-hackathon-server/requirements.txt
+RUN pip install -r /opt/open-hackathon/open-hackathon-server/requirements.txt -i https://pypi.tuna.tsinghua.edu.cn/simple --no-cache-dir
+
+COPY ./open-hackathon-server /opt/open-hackathon/open-hackathon-server
+
+# TODO use yaml config
+RUN mv /opt/open-hackathon/open-hackathon-server/src/hackathon/config_docker.py /opt/open-hackathon/open-hackathon-server/src/hackathon/config_sample.py
```
Comment (juniwang) [inline deploy/hackathon-docker/Dockerfile-server:28]: rename to `config.py`

File: deploy/hackathon-docker/Dockerfile-server [hunk 2]
```diff
@@ -10,24 +10,24 @@ FROM python:3.6-alpine
...
 
 WORKDIR /opt/open-hackathon/open-hackathon-server/src
+CMD ["gunicorn", "-w", "5", "-b", "0.0.0.0:15000", "hackathon:app"]
 
-ENTRYPOINT ./opt/open-hackathon/docker-entrypoint-server.sh
+EXPOSE 15000
```
Comment (juniwang) [inline deploy/hackathon-docker/Dockerfile-server:33]: Suggest we use 443 in future.

Comment (Coderhypo) [reply deploy/hackathon-docker/Dockerfile-server:33]: 这个只是容器端口，可以在部署时来通过 service 来重新映射。如果是 https 的话，不需要改容器端口，只需要配置好 Ingress 规则，k8s 自己配置类似 nginx 的 proxy_pass 的事情。

Comment (juniwang) [reply deploy/hackathon-docker/Dockerfile-server:30]: ok

**Comment by juniwang** [comment]
`config_docker.py`删掉了？里面从环境变量读的参数咋弄？

**Comment by Coderhypo** [comment]
> `config_docker.py`删掉了？里面从环境变量读的参数咋弄？

没注意 `config.py` 命中 gitignore 规则了，修改了下 gitignore

File: deploy/hackathon-docker/Dockerfile-server (hunk 1 above)
Comment (99Kies) [reply deploy/hackathon-docker/Dockerfile-server]: server端判断有误，现在只认可config_sample

**Comment by 99Kies** [comment]
优化Dockerfile-client
```
# -*- coding: utf-8 -*-
#
# This file is covered by the LICENSING file in the root of this project.
#

### Dockerfile for open-hacnathon(client)
### based on python official image

FROM python:3.7-alpine

USER root

#logs
RUN mkdir -p /var/log/open-hackathon/
#client
RUN mkdir -p /opt/open-hackathon/open-hackathon-client

COPY ./open-hackathon-client /opt/open-hackathon/open-hackathon-client

#pip
RUN pip install -r /opt/open-hackathon/open-hackathon-client/requirements.txt

#config
RUN mv /opt/open-hackathon/open-hackathon-client/src/client/config_docker.py /opt/open-hackathon/open-hackathon-client/src/client/config.py

#init
COPY ./deploy/hackathon-docker/docker-entrypoint-client.sh /opt/open-hackathon

RUN chmod 777 /opt/open-hackathon/docker-entrypoint-client.sh && \
    dos2unix /opt/open-hackathon/docker-entrypoint-client.sh

EXPOSE 80

ENTRYPOINT ./opt/open-hackathon/docker-entrypoint-client.sh

```

**Comment by 99Kies** [comment]
> 优化Dockerfile-client
> 
> ```
> # -*- coding: utf-8 -*-
> #
> # This file is covered by the LICENSING file in the root of this project.
> #
> 
> ### Dockerfile for open-hacnathon(client)
> ### based on python official image
> 
> FROM python:3.7-alpine
> 
> USER root
> 
> #logs
> RUN mkdir -p /var/log/open-hackathon/
> #client
> RUN mkdir -p /opt/open-hackathon/open-hackathon-client
> 
> COPY ./open-hackathon-client /opt/open-hackathon/open-hackathon-client
> 
> #pip
> RUN pip install -r /opt/open-hackathon/open-hackathon-client/requirements.txt
> 
> #config
> RUN mv /opt/open-hackathon/open-hackathon-client/src/client/config_docker.py /opt/open-hackathon/open-hackathon-client/src/client/config.py
> 
> #init
> COPY ./deploy/hackathon-docker/docker-entrypoint-client.sh /opt/open-hackathon
> 
> RUN chmod 777 /opt/open-hackathon/docker-entrypoint-client.sh && \
>     dos2unix /opt/open-hackathon/docker-entrypoint-client.sh
> 
> EXPOSE 80
> 
> ENTRYPOINT ./opt/open-hackathon/docker-entrypoint-client.sh
> ```

**Comment by 99Kies** [comment]
这样子修改之后 docker-compose up应该不能启动了吧，要不kubernetes的内容单独开一个文件夹

**Pull request merged**